java-checks/src/main/java/org/sonar/java/checks/ClassWithOnlyStaticMethodsInstantiationCheck.java

```diff
@@ -84,13 +85,20 @@ private static boolean hasOnlyStaticMethodsAndFields(Symbol.TypeSymbol newClassT
         return false;
       }
     }
-    return superClassHasOnlyStaticMethods(newClassTypeSymbol);
+    return superTypesHaveOnlyStaticMethods(newClassTypeSymbol);
   }
 
-  private static boolean superClassHasOnlyStaticMethods(Symbol.TypeSymbol newClassTypeSymbol) {
+  private static boolean superTypesHaveOnlyStaticMethods(Symbol.TypeSymbol newClassTypeSymbol) {
     Type superClass = newClassTypeSymbol.superClass();
     if (superClass != null && !superClass.is("java.lang.Object")) {
-      return hasOnlyStaticMethodsAndFields(superClass.symbol());
+      if (!hasOnlyStaticMethodsAndFields(superClass.symbol())) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Merge this if statement with the enclosing one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1066)